### PR TITLE
Fix stacklocked error for succeeding requests in debug mode

### PR DESCRIPTION
**Pull request by gafrom**
Hi there.

This addresses the [issue #189](https://github.com/intridea/oauth2/issues/189).

Tests added and they pass.

### CODE REVIEW

**Comment by JasonLunn** [comment]
👍

**Comment by beingben** [comment]
This is a much needed patch - please review and give feedback if it's not the right thing to do.

**Comment by djberg96** [comment]
@sferik Any chance of getting this merged?

**Comment by xtagon** [comment]
Looks good to me (once the new conflicts are fixed)

**Comment by josephpage** [comment]
Great job for this old and annoying bug !
#331 already solved this issue, but without testing.
Only the test should be merged.
@gafrom Can you please solve the conflict by keeping only the test ?

**Comment by gafrom** [comment]
@josephpage Conflicts solved, thanks 🎄

File: spec/oauth2/client_spec.rb [hunk 1]
```diff
@@ -159,6 +159,23 @@
...
+
+      capture_output do
+        # first request (always goes smoothly)
+        subject.request(:get, '/success')
+      end
+
+      expect do
+        # second request (used to throw Faraday::RackBuilder::StackLocked)
+        subject.request(:get, '/success')
+      end.not_to raise_error
```
Comment (syakovyn) [inline spec/oauth2/client_spec.rb]: Need `end.not_to raise Faraday::RackBuilder::StackLocked` here. `raise_error` doesn't fail on `Faraday::RackBuilder::StackLocked` for some reason.

**Pull request merged**